homeassistant/components/home_plus_control/api.py

```diff
@@ -0,0 +1,352 @@
+                except Exception as err:  # pylint: disable=broad-except
+                    self.logger.error(
+                        "Error encountered when updating plant topology for plant %s: %s [%s]",
+                        plant.id,
+                        err,
+                        type(err),
+                    )
+                else:
+                    # If all goes well, we update the last check time
+                    self._last_check[
+                        CONF_PLANT_TOPOLOGY_UPDATE_INTERVAL
+                    ] = time.monotonic()
+
+            if self._should_check(
+                CONF_MODULE_STATUS_UPDATE_INTERVAL,
+                self._refresh_intervals[CONF_MODULE_STATUS_UPDATE_INTERVAL],
+            ):
+                self.logger.debug("API update of module status for plant %s.", plant.id)
+                try:
+                    await plant.update_module_status()  # Call the API
+                except Exception as err:  # pylint: disable=broad-except
+                    self.logger.error(
+                        "Error encountered when updating plant module status for plant %s: %s [%s]",
+                        plant.id,
+                        err,
+                        type(err),
+                    )
+                else:
+                    # If all goes well, we update the last check time
+                    self._last_check[
+                        CONF_MODULE_STATUS_UPDATE_INTERVAL
+                    ] = time.monotonic()
+
+        return self._update_modules()
+
+    def _should_check(self, check_type, period):
+        """Return True if the current monotonic time is > the last check time plus a fixed period.
+
+        Args:
+            check_type (str): Type that identifies the timer that has to be used
+            period (float): Number of fractional seconds to add to the last check time
+        """
+        current_time = time.monotonic()
+        if (
+            self._last_check[check_type] == -1
+            or current_time > self._last_check[check_type] + period
+        ):
+            self.logger.debug(
+                "Last check time (%.2f) exceeded by more than %.2f sec - monotonic time %.2f",
+                self._last_check[check_type],
+                period,
+                current_time,
+            )
+            return True
+        return False
+
+    def _update_modules(self):
+        """Update the switch modules based on the collected information in the plant object.
+
+        Returns:
+            dict: Dictionary of switch modules across all of the plants.
+        """
+        for plant in self._plants.values():
+            # Loop through the modules in the plant and we only keep the ones that are "interactive"
+            # and can be represented by a switch, i.e. power outlets, micromodules
+            # and light switches. All other modules are discarded/ignored.
+            current_module_ids = []
```

Comment:
Make this a set. Then we can easily do different membership operations eg intersection.
```suggestion
            current_module_ids = set()
```